consumer.go

```diff
@@ -547,9 +547,11 @@ func (child *partitionConsumer) parseRecords(batch *RecordBatch) ([]*ConsumerMes
 		}
 	}
 
-	if incomplete || len(messages) == 0 {
+	if incomplete {
 		return nil, ErrIncompleteResponse
 	}
+
+	child.offset = batch.FirstOffset + int64(batch.LastOffsetDelta) + 1
```

Comment:
Should this be guarded by a check that this actually advances the offset? If it doesn't and/or if messages were present which don't line up, we should probably still return an error of some sort.

Reply:
Good call, I added a check for that now. The check is after the assignment rather than guarding it as L554 should not advance the offset when processing a normal non-compacted topic.